cookbook/import_export/create-custom-step.rst

```diff
@@ -15,36 +15,92 @@ We will begin by creating a NotifyStep with its configuration and a doExecute me
    :language: php
    :linenos:
-   :lines: 1-4,10-
 
-Create our Step Element
------------------------
+Define the Step as a service
+----------------------------
 
-Then, we implement a Step Element, in our case, a handler responsible for sending a ping request:
+Add your step element to ``steps.yml`` to ensure that it is loaded and processed by ``DependencyInjection/AcmeNotifyConnectorExtension``:
+
+.. literalinclude:: ../../src/Acme/Bundle/NotifyConnectorBundle/Resources/config/steps.yml
+   :language: yaml
+   :linenos:
+
+Configure our new Job
+---------------------
+
+In ``Resources/config/jobs.yml``, we use a first step to export products in csv and we configure the second one to send a notification:
+
+.. literalinclude:: ../../src/Acme/Bundle/NotifyConnectorBundle/Resources/config/jobs.yml
+   :language: yaml
+   :linenos:
+
+Configure the job parameters
+----------------------------
+
+In order to setup the form we use a form configuration provider.
+
+First of all you need to create a new class that will contains the job parameters. This class should implement the following interfaces:
+- ``ConstraintCollectionProviderInterface``
+- ``DefaultValuesProviderInterface``
+- ``FormConfigurationProviderInterface``
 
-.. literalinclude:: ../../src/Acme/Bundle/NotifyConnectorBundle/Handler/CurlHandler.php
+Here we decorate the ``ProductCsvExport`` classes in order to retrieve the proper form export configuration.
+
+.. literalinclude:: ../../src/Acme/Bundle/NotifyConnectorBundle/JobParameters/ProductCsvExportNotify.php
    :language: php
    :linenos:
-   :lines: 1-3,9-
+   :lines: 31-39
 
-Define the Step Element as Service
-----------------------------------
+Then we add the form configuration for our new ``url`` field.
 
-Add your step element to ``steps.yml`` to ensure that it is loaded and processed by ``DependencyInjection/AcmeNotifyConnectorExtension``:
+.. literalinclude:: ../../src/Acme/Bundle/NotifyConnectorBundle/JobParameters/ProductCsvExportNotify.php
+   :language: php
+   :linenos:
+   :lines: 56-69
 
-.. literalinclude:: ../../src/Acme/Bundle/NotifyConnectorBundle/Resources/config/steps.yml
+In order to validate the job parameters we need to define some constraints. Here we want a valid url as input.
+
+.. literalinclude:: ../../src/Acme/Bundle/NotifyConnectorBundle/JobParameters/ProductCsvExportNotify.php
+   :language: php
+   :linenos:
+   :lines: 44-51
+
+We also need to add the default values to our job form.
+
+.. literalinclude:: ../../src/Acme/Bundle/NotifyConnectorBundle/JobParameters/ProductCsvExportNotify.php
+   :language: php
+   :linenos:
+   :lines: 74-79
+
+And don't forget the support method with your job name ``csv_product_export_notify``.
+
+.. literalinclude:: ../../src/Acme/Bundle/NotifyConnectorBundle/JobParameters/ProductCsvExportNotify.php
+   :language: php
+   :linenos:
+   :lines: 85-88
+
+And don't forget to declare this class as a service, with the proper job name as parameter:
```

Comment:
"And don't forget" is used twice, maybe just "Add the support method with your [...]" and for the second one "Declare this class as a service" to keep it simple ?